### PR TITLE
Update codestyle, php version, tests, add interface

### PR DESCRIPTION
**Pull request by Jeroeny**
This library hasn't been updated in a while. I'm not sure if that's intentional, but I've taken an attempt to modernize this library a bit. I hope an update for this library can be considered. If this is not the right approach of some changes should be left out / split in other PRs, let me know.

I've added an interface,  which would be nice to have for dependency injection and mocking in tests. Updated PHPUnit and the tests. Used the last maintained PHP version as minimum and used types where possible. Applied the PSR codestyle. Added GitHub actions configuration.

### CODE REVIEW

File: src/Exception/ConfigurationException.php [hunk 1]
```diff
@@ -3,37 +3,33 @@
...
      * Client instance that threw the exception
-     * @var Client
      */
-    protected $instance;
-
+    protected StatsDClient $instance;
 
     /**
      * Create new instance
      * @param Client $instance Client instance that threw the exception
```
Comment (andrew-demb) [inline src/Exception/ConfigurationException.php:20]:
```suggestion
     * @param StatsDClient $instance Client instance that threw the exception
```

File: src/StatsDClient.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace League\StatsD;
+
+use League\StatsD\Exception\ConnectionException;
+
+/**
+ * StatsD Client Class
```
Comment (andrew-demb) [inline src/StatsDClient.php:8]:
```suggestion
 * StatsD Client Interface
```

File: src/Client.php [hunk 3]
```diff
@@ -2,139 +2,92 @@
...
+    /** Instance instances array */
+    protected static array $instances = [];
 
-    /**
-     * Instance instances array
-     * @var array
-     */
-    protected static $instances = array();
+    /** Instance ID */
+    protected string $instanceId;
```
Comment (andrew-demb) [inline src/Client.php:19]: It is can be considered as BC break (previous name was a `instance_id`)

Comment (andrew-demb) [reply src/Client.php:19]: But adding a property typehints can be considered as BC break too, so this renaming can be approved too.

:thumbsup:

**Review by andrew-demb — COMMENTED** [review]
I'm not a maintainer, but also can take a look for changes.

Overall I think PR is fine :thumbsup:
Probably major release will be required

**Comment by Jeroeny** [comment]
@marcqualie would you be willing to take a look?

**Comment by xvilo** [comment]
This PR looks good to me 👍

**Review by colinodell — APPROVED** [review]

**Comment by Jeroeny** [comment]
@colinodell I think this can be merged now?

**Comment by colinodell** [comment]
I'll get it merged sometime this week (possibly this weekend).  I'd also like to review the other outstanding PRs and see if we can include them too :)

File: composer.json [hunk 4]
```diff
@@ -16,8 +16,12 @@
             "role": "Developer"
         }
     ],
+    "require": {
+        "php": ">=7.4"
```
Comment (nyamsprod) [inline composer.json:20]: Maybe best to do "php": "^7.4 || ^8.0"

**Comment by colinodell** [comment]
[Version 2.0.0](https://github.com/thephpleague/statsd/releases/tag/2.0.0) has been released! Thanks so much for your help with this :)

**Pull request merged**